src/shared/json.c

```diff
@@ -549,9 +549,33 @@ static void json_variant_copy_source(JsonVariant *v, JsonVariant *from) {
         v->source = json_source_ref(from->source);
 }
 
+static int _json_variant_array_put_element(JsonVariant *array, JsonVariant *element) {
+        JsonVariant *w = array + 1 + array->n_elements;
```

Comment:
`assert(array)`

Reply:
I wanted to use ASSERT_PTR(), but it'd need to be present twice, because the order of evaluation of operands is not defined. So I added `assert(array);` in the first line instead.